### PR TITLE
[Refactor] Refactor replication to add some replication related be configs

### PR DESCRIPTION
**Pull request by xiangguangyxg**
## What I'm doing:

1. Add some replication related be configs.
2. Separate thread pool of remote snapshot and replicate snapshot.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by wyb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/replication/ReplicationJob.java [hunk 1]
```diff
@@ -982,11 +982,13 @@ private void removeRunningTasks() {
     }
 
     private boolean isAllTaskFinished() {
+        LOG.info("Replication job state: {}, finished tasks: {}/{}, database id: {}, table id: {}, transaction id: {}",
+                state, finishedTasks.size(), taskNum, databaseId, tableId, transactionId);
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/replication/ReplicationJob.java:986]: this info log can't determine the the following true or false result. will the runningTasks.size() is necessary as well in this log?

File: be/src/storage/replication_utils.cpp [hunk 2]
```diff
@@ -277,9 +280,9 @@ Status ReplicationUtils::download_remote_snapshot(
         }
 
         total_file_size += file_size;
-        uint64_t estimate_timeout_sec = file_size / config::download_low_speed_limit_kbps / 1024;
-        if (estimate_timeout_sec < config::download_low_speed_time) {
-            estimate_timeout_sec = config::download_low_speed_time;
+        uint64_t estimate_timeout_sec = file_size / config::replication_min_speed_limit_kbps / 1024;
```
Comment (kevincai) [inline be/src/storage/replication_utils.cpp:283]: is there configvar validation? try to avoid divide-0 error.

Comment (xiangguangyxg) [reply be/src/storage/replication_utils.cpp:283]: no, I will add

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by xiangguangyxg** [comment]
@Mergify backport branch-3.3

**Comment by xiangguangyxg** [comment]
@Mergify backport branch-3.2

**Comment by xiangguangyxg** [comment]
@Mergify backport branch-3.1

**Pull request merged**